Sources/StreamChat/WebSocketClient/EventMiddlewares/MemberEventMiddleware.swift

```diff
@@ -28,33 +32,64 @@ struct MemberEventMiddleware: EventMiddleware {
                 
                 // We remove the member from the channel
                 channel.members.remove(member)
+                if let membership = channel.membership, membership.user.id == event.user.id {
+                    channel.membership = nil
+                }
                 
                 // If there are any MemberListQueries observing this channel,
                 // we need to update them too
                 member.queries.removeAll()
                 
             case let event as NotificationAddedToChannelEventDTO:
-                _ = try session.saveChannel(payload: event.channel, query: nil)
+                let channel = try session.saveChannel(payload: event.channel, query: nil)
+                let member = try session.saveMember(payload: event.member, channelId: event.channel.cid)
+                channel.membership = member
+                
+                insertMemberToMemberListQueries(channel, member)
                 
             case let event as NotificationRemovedFromChannelEventDTO:
                 guard let channel = session.channel(cid: event.cid) else {
                     // No need to throw ChannelNotFound error here
-                    log.debug("Channel \(event.cid) not found for MemberRemovedEventDTO")
+                    log.debug("Channel \(event.cid) not found for NotificationRemovedFromChannelEventDTO")
                     break
                 }
                 
                 guard let member = channel.members.first(where: { $0.user.id == event.member.user.id }) else {
                     // No need to throw MemberNotFound error here
-                    log.debug("Member \(event.member.user.id) not found for MemberRemovedEventDTO")
+                    log.debug("Member \(event.member.user.id) not found for NotificationRemovedFromChannelEventDTO")
                     break
                 }
                 
                 // We remove the member from the channel
                 channel.members.remove(member)
+                if let membership = channel.membership, membership.user.id == event.member.user.id {
+                    channel.membership = nil
+                }
```

Comment:
I would always reset the membership 🙂 Especially when `membership.user.id != event.member.user.id`. If the user id does not match the membership it means we didn't properly clean up the database from the previous user and in this case we'd better recover the local date to be valid for the current user

```suggestion
                channel.membership = nil
```